lib/membrane/rtp/demuxer/jitter_buffer.ex

```diff
@@ -0,0 +1,262 @@
+          }
+
+  defmodule State do
+    @moduledoc false
+
+    @type t :: %__MODULE__{
+            buffer_store: RTP.JitterBuffer.BufferStore.t(),
+            clock_rate: non_neg_integer() | nil,
+            latency: Membrane.Time.t() | nil,
+            initial_latency_waiting: boolean(),
+            initialization_time: Membrane.Time.t(),
+            max_latency_timer: reference() | nil,
+            timestamp_base: Membrane.Time.t(),
+            previous_timestamp: Membrane.Time.t()
+          }
+
+    @enforce_keys [
+      :initial_latency_waiting,
+      :initialization_time,
+      :timestamp_base,
+      :previous_timestamp
+    ]
+
+    defstruct @enforce_keys ++
+                [
+                  buffer_store: %RTP.JitterBuffer.BufferStore{},
+                  clock_rate: nil,
+                  latency: nil,
+                  max_latency_timer: nil
+                ]
+  end
+
+  @spec initialize_state_with_new_rtp_stream(
+          ExRTP.Packet.t(),
+          pad_options() | nil,
+          RTP.PayloadFormat.payload_type_mapping()
+        ) :: State.t()
+  def initialize_state_with_new_rtp_stream(packet, pad_options, payload_type_mapping) do
+    {clock_rate, latency, initial_latency_waiting} =
+      case pad_options do
+        nil ->
+          {nil, nil, true}
+
+        %{use_jitter_buffer: false} = pad_options ->
+          %{clock_rate: clock_rate} =
+            RTP.PayloadFormat.resolve(
+              payload_type: packet.payload_type,
+              clock_rate: pad_options.clock_rate,
+              payload_type_mapping: payload_type_mapping
+            )
+
+          {clock_rate, 0, false}
+
+        %{use_jitter_buffer: true} = pad_options ->
+          %{clock_rate: clock_rate} =
+            RTP.PayloadFormat.resolve(
+              payload_type: packet.payload_type,
+              clock_rate: pad_options.clock_rate,
+              payload_type_mapping: payload_type_mapping
+            )
+
+          Process.send_after(
+            self(),
+            {:initial_latency_passed, packet.ssrc},
+            Membrane.Time.as_milliseconds(pad_options.jitter_buffer_latency, :round)
+          )
+
+          {clock_rate, pad_options.jitter_buffer_latency, true}
+      end
+
+    %State{
+      initial_latency_waiting: initial_latency_waiting,
+      initialization_time: Membrane.Time.monotonic_time(),
+      timestamp_base: packet.timestamp,
+      previous_timestamp: packet.timestamp,
+      clock_rate: clock_rate,
+      latency: latency
+    }
+  end
+
+  @spec complete_state_initialization_with_pad_options(
+          State.t(),
+          pad_options(),
+          RTP.payload_type(),
+          RTP.ssrc(),
+          RTP.PayloadFormat.payload_type_mapping()
+        ) :: State.t()
+  def complete_state_initialization_with_pad_options(
+        jitter_buffer_state,
+        pad_options,
+        payload_type,
+        ssrc,
+        payload_type_mapping
```

Comment:
We already pass these to `initialize_state_with_new_rtp_stream`, can we keep them in state not to pass them again?